src/rules/select-style.ts

```diff
@@ -0,0 +1,62 @@
+import { ESLintUtils, TSESTree } from '@typescript-eslint/experimental-utils'
+
+import { docsUrl, pipeableSelect, storeSelect } from '../utils'
+
+export const ruleName = 'select-style'
+
+export const methodSelectMessageId = 'storeSelect'
+export const operatorSelectMessageId = 'operatorSelect'
+
+export type MessageIds =
+  | typeof methodSelectMessageId
+  | typeof operatorSelectMessageId
+
+export const OPERATOR = 'operator'
+export const METHOD = 'method'
+
+type Options = [string]
+
+export default ESLintUtils.RuleCreator(docsUrl)<Options, MessageIds>({
+  name: ruleName,
+  meta: {
+    type: 'problem',
+    docs: {
+      category: 'Possible Errors',
+      description: `Selectors can be used either with 'select' as a pipeable operator or as a method`,
+      recommended: false,
+    },
+    schema: [
+      {
+        enum: [OPERATOR, METHOD],
+      },
+    ],
+    messages: {
+      [methodSelectMessageId]:
+        'Selectors should be used with select method: this.store.select(selector)',
+      [operatorSelectMessageId]:
+        'Selectors should be used with the pipeable operator: this.store.pipe(select(selector))',
+    },
+  },
+  defaultOptions: [OPERATOR],
+  create: (context) => {
+    const mode: string = context.options[0] || OPERATOR
```

Comment:
Is it needed to fallback to `OPERATOR`? I thought this would  be handled by the default options config.
Perhaps for this to work, we should use the following syntax (taken from [angular eslint](https://github.com/angular-eslint/angular-eslint/blob/master/packages/eslint-plugin/src/rules/component-class-suffix.ts#L48)).

```suggestion
  create(context, [options]) {
    const { mode } = options;
```